srml/assets/src/lib.rs

```diff
@@ -14,7 +14,135 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
+//! # Assets Module
+//! <!-- Original author of paragraph: @gavofyork -->
+//!
 //! A simple, secure module for dealing with fungible assets.
+//!
+//! ## Overview
+//!
+//! The assets module provides functionality for asset management of fungible asset classes with a fixed supply, including:
+//!
+//! * Asset Issuance
+//! * Asset Transfer
+//! * Asset Destruction
+//!
+//! To use it in your module, you need to implement the assets [`Trait`].<br />
+//! The supported dispatchable functions are documented in the [`Call`] enum.
+//!
+//! ## Terminology
+//!
+//! * **Asset issuance:** The process of an account issuing a total fixed supply of a new asset class.
+//! * **Asset transfer:** The process of an account transfering units of their holding of an asset to a recipient.
+//! * **Asset destruction:** The process of an account destroying their entire holding of an asset.
```

Comment:
```suggestion
//! * **Asset destruction:** The process of an account removing its entire holding of an asset.
```

Reply:
I originally went with `destroy` to keep it simple since that's also the name of the public function.

Reply:
Exactly, I wouldn't use a word to define itself.

Reply:
Either way, account in this case is always singular so it should say "its" not "their"

Reply:
Good point!